process/block/sovereignChainBlock.go

```diff
@@ -261,6 +266,54 @@ func (scbp *sovereignChainBlockProcessor) CreateBlock(initialHdr data.HeaderHand
 	return initialHdr, &block.Body{MiniBlocks: miniBlocks}, nil
 }
 
+// We should call this func only on ProcessBlock for all participants.
+// No need to call it on CreateBlock for epoch start processing.
+func (scbp *sovereignChainBlockProcessor) updateEpochStartHeader(header data.SovereignChainHeaderHandler) (*block.Economics, error) {
+	sovHeaderHandler, castOk := header.(*block.SovereignChainHeader)
+	if !castOk {
+		return nil, fmt.Errorf("%w in sovereignChainBlockProcessor.updateEpochStartHeader", process.ErrWrongTypeAssertion)
+	}
+
+	sw := core.NewStopWatch()
+	sw.Start("createEpochStartForSovereignBlock")
+	defer func() {
+		sw.Stop("createEpochStartForSovereignBlock")
+		log.Debug("epochStartHeaderDataCreation", sw.GetMeasurements()...)
+	}()
+
+	totalAccumulatedFeesInEpoch := big.NewInt(0)
+	totalDevFeesInEpoch := big.NewInt(0)
+	currentHeader := scbp.blockChain.GetCurrentBlockHeader()
+	if !check.IfNil(currentHeader) && !currentHeader.IsStartOfEpochBlock() {
```

Comment:
`updateEpochStartHeader` is called only `if shardHeader.IsStartOfEpochBlock()` in ProcessBlock, so this `if` is never called.
I see you set the fees in `applyBodyToHeader` at the end of `processSovereignBlockTransactions` function.

Reply:
The namings and processing flow might create this confusions.
`currentHeader := scbp.blockChain.GetCurrentBlockHeader()` -> `currentHeader` represents the **current committed and agreed upon header inside the network** 
`updateEpochStartHeader(header data.SovereignChainHeaderHandler) ` -> `header` represents the **current header that is being created to be added on the of the `currentHeader` **

So the code is correct, that code is actually being called

Reply:
`processSovereignBlockTransactions` is not called for epoch start block processing, see this code:

```go
	if shardHeader.IsStartOfEpochBlock() {
		err = scbp.processEpochStartMetaBlock(shardHeader, body)
		if err != nil {
			return nil, nil, err
		}

		body, err = scbp.applyBodyToHeader(shardHeader, body)
		if err != nil {
			return nil, nil, err
		}

		// finish processing here, only process epoch start
		return shardHeader, body, nil
	}

```